src/components/QueryResultTable/utils/getColumnWidth.ts

```diff
@@ -0,0 +1,21 @@
+import type {KeyValueRow} from '../../../types/api/query';
+
+export const MAX_COLUMN_WIDTH = 600;
+export const HEADER_PADDING = 20;
+
+export const getColumnWidth = ({data, name}: {data?: KeyValueRow[]; name: string}) => {
+    let maxColumnContentLength = name.length;
+
+    if (data) {
+        for (const row of data) {
+            const cellLength = row[name] ? String(row[name]).length : 0;
+            maxColumnContentLength = Math.max(maxColumnContentLength, cellLength);
+
+            if (maxColumnContentLength * 10 + HEADER_PADDING >= MAX_COLUMN_WIDTH) {
+                return MAX_COLUMN_WIDTH;
+            }
+        }
+    }
+
+    return Math.min(maxColumnContentLength * 10 + HEADER_PADDING, MAX_COLUMN_WIDTH);
```

Comment:
it seems we may skip this check, cause it's already done: https://github.com/ydb-platform/ydb-embedded-ui/pull/1384/files#diff-372ca73fb142649c09d5b178debd37c1c6507e47eb54bf23cbf447d7f4904e1bR14